test/unit/connectors/CustomsNotificationConnectorSpec.scala

```diff
@@ -0,0 +1,140 @@
+
+  private val validConfigs: Map[String, Any] = Map(
+    ("microservice.services.customs-notification.host" -> serviceHost),
+    ("microservice.services.customs-notification.port" -> "80"),
+    ("microservice.services.customs-notification.context" -> serviceContext),
+    ("microservice.services.customs-notification.bearer-token" -> basicAuthToken))
+
+  private val expectedUrl = s"http://$serviceHost:80$serviceContext"
+  private val expectedHeaders: Map[String, String] = Map(
+    ("X-CDS-Client-ID" -> customsNotification.clientSubscriptionId),
+    ("X-Conversation-ID" -> customsNotification.conversationId.toString),
+    (CONTENT_TYPE -> s"${MimeTypes.XML}; charset=UTF-8"),
+    (ACCEPT -> MimeTypes.XML),
+    (AUTHORIZATION -> s"Basic $basicAuthToken"))
+
+  override protected def beforeEach() {
+    reset(mockLogger, mockHttpClient, mockConfigs)
+  }
+
+  "CustomsNotificationConnector" should {
+
+    "fail to start when host is not configured" in {
+      val caught = intercept[RuntimeException] {
+        new CustomsNotificationConnector(mockHttpClient, mockLogger, servicesConfigFrom(Map[String, String]()))
+      }
+
+      caught.getMessage shouldBe "Could not find config customs-notification.host"
+    }
+
+    "fail to start when port is not configured" in {
+      val caught = intercept[RuntimeException] {
+        new CustomsNotificationConnector(mockHttpClient, mockLogger, servicesConfigFrom(Map("microservice.services.customs-notification.host" -> "some")))
+      }
+
+      caught.getMessage shouldBe "Could not find config customs-notification.port"
+    }
+
+    "fail to start when Context is not configured" in {
+      val caught = intercept[IllegalStateException] {
+        new CustomsNotificationConnector(mockHttpClient, mockLogger, servicesConfigFrom(validConfigs - "microservice.services.customs-notification.context"))
+      }
+
+      caught.getMessage shouldBe "Configuration error - customs-notification.context not found."
+    }
+
+    "fail to start when auth token is not configured" in {
+      val caught = intercept[IllegalStateException] {
+        new CustomsNotificationConnector(mockHttpClient, mockLogger, servicesConfigFrom(validConfigs - "microservice.services.customs-notification.bearer-token"))
+      }
+
+      caught.getMessage shouldBe "Configuration error - customs-notification.basicAuthToken not found."
+    }
+
+    "make correct request" in {
+      when(mockHttpClient.POSTString[HttpResponse](any(), any(), any())(any(), any(), any()))
+        .thenReturn(Future.successful(mock[HttpResponse]))
+
+      await(connector.send(customsNotification))
+
+      verify(mockHttpClient).POSTString(
+        ameq(expectedUrl),
+        ameq(s"""<?xml version="1.0" encoding="UTF-8" standalone="yes"?>""" + customsNotification.payload.toString()),
+        ameq(expectedHeaders.toSeq)
+      )(any(), any(), any())
+    }
+
+    "log the error in case request to notification service fails" in {
```

Comment:
we have constant in [emulatedServiceFailure](https://github.com/hmrc/customs-declarations/blob/master/test/util/TestData.scala#L57) for this

Reply:
that is unsupported operation exception, I wanted to make sure that it works for any runtime exception hence the parent RuntimeException is used.